src/interceptors/axios/utils/index.ts

```diff
@@ -9,3 +19,113 @@ export const getPublicKeyFromContainer = async (containerUrl: string): Promise<s
 
   return data.publicKey;
 };
+
+/**
+ * A Transform stream that encrypts data using the Polaris SDK.
+ */
+export class EncryptStream extends Transform {
+  private sdk: PolarisSDK;
+  constructor(sdk: PolarisSDK) {
+    super();
+    this.sdk = sdk;
+  }
+
+  /**
+   * Encrypts the given data.
+   * @param data - The data to encrypt as a string.
+   * @returns A promise that resolves to the encrypted data as a Buffer.
+   * @throws An error if encryption fails.
+   */
+  async encrypt(data: string): Promise<Buffer> {
+    try {
+      const publicKey = await this.sdk.getPublicKey();
```

Comment:
We should cache the public key in the object (best in the constructor) as for some KeyHandlers it may cause an HTTP request to be made